### DISCUSSION TITLE
Dockerless Install

### DISCUSSION THREAD

**Discussion by chrishobcroft** [nextcloud/all-in-one#2410] [Ideas] [answered]
As a user, I want to run AIO using `systemd` on Linux, i.e. without requiring the added complexity of running `docker`.

**Comment by Zoey2936** [comment, marked as answer]
Not sure what you expect as answer, but that would require a new project and from my perspective doing all this using systemd would be unmaintainable

**Comment by chrishobcroft** [comment]
I'm not sure what you mean "new project" but it could be as simple as publishing instructions to compile a binary from source, and run as a simple process.

Requiring docker introduces an extra dependency which may be convenient but certainly isn't necessary.

**Reply by Zoey2936** [reply]
I'm not sure, but I think you have no clue how docker works, am I wrong?
You can't build AIO from source and run it using systemd, that is impossible.

**Comment by chrishobcroft** [comment]
You are wrong

**Reply by Zoey2936** [reply]
what do you mean?

**Reply by chrishobcroft** [reply]
You asked me if you were wrong. I told you that you are wrong.

**Reply by chrishobcroft** [reply]
Clearly this conversation isn't going anywhere.

**Reply by Zoey2936** [reply]
Maybe, but I don't understand what you expect, Docker images can't run with systemd and if you want to install everything manually, have a look in the docker files and start scripts and configs inside the Containers folder and try to install everything manually. you will need to adjust everything for your OS and I would say that is very hard, better read the documentations of the software built-in here and install it this way.

You don't need to answer, but what is your problem with docker?

**Reply by chrishobcroft** [reply]
It adds an unnecessary additional layer of complexity. I am all for subtracting complexity. It also introduces a dependency on a private company, and I am all for minimising dependencies on private companies.

**Comment by szaimen** [comment]
Just to support Zoey here: this project is completely built around and relying on docker. So removing docker would mean a complete rewrite which we obviously will not and cannot do.

However podman would in theory be a possible runtime as well, if they would be 100% compatible to the Docker API.

**Comment by jennydaman** [comment]
Podman as an alternative to Docker works, has good integration with systemd ([read this article](https://www.redhat.com/sysadmin/quadlet-podman)) and its rootless mode may alleviate your concerns about Docker. 

If you're interested, my working setup can be found here: https://github.com/nextcloud/all-in-one/discussions/471#discussioncomment-6687371
